src/main/java/com/google/devtools/build/lib/rules/cpp/LibrariesToLinkCollector.java

```diff
@@ -332,6 +332,14 @@ private void addDynamicInputLinkOptions(
 
       rpathRootsForExplicitSoDeps.add(
           rpathRoot + dotdots + libDir.relativeTo(commonParent).getPathString());
+
+      // Unless running locally, libraries will be available under the root relative path, so we
+      // should add that to the rpath as well.
+      if (inputArtifact.getRootRelativePathString().startsWith("_solib_")) {
+        PathFragment artifactPathUnderSolib = inputArtifact.getRootRelativePath().subFragment(1);
+        rpathRootsForExplicitSoDeps.add(
```

Comment:
Thank you for getting to this!

Your understanding is the same as mine: A is where the symlink exists locally with a transition, and B is where it exists otherwise (no transition or not running locally). As far as I remember, B doesn't exist when running locally.

Reply:
Do you know why A) is not created with a transition running remotely?

Reply:
`k8-fastbuild...` (both with and without transitions) aren't created for remote execution or under the runfiles directory, and the correct versions of shared libraries are found under the solib directory instead.